packages/vite/src/node/optimizer/esbuildDepPlugin.ts

```diff
@@ -258,11 +262,20 @@ module.exports = Object.create(new Proxy({}, {
       if (isRunningWithYarnPnp) {
         build.onResolve(
           { filter: /.*/ },
-          async ({ path, importer, kind, resolveDir }) => ({
-            // pass along resolveDir for entries
-            path: await resolve(path, importer, kind, resolveDir)
-          })
+          async ({ path: id, importer, kind, resolveDir, namespace }) => {
+            const resolved = await resolve(
+              id,
+              importer,
+              kind,
+              // pass along resolveDir for entries
+              namespace === 'dep' ? resolveDir : undefined
```

Comment:
browser-external proxy from #8338 fixes build-time crashed, but something has changed elsewhere and this workaround doesn't work anymore 🥲 Will have to look into the changes since 2.9.9

Reply:
Oh, it does work for external dependencies

https://github.com/swandir/vite-browser-external-pnp/blob/1551d40a3cfb8f59915e43f23a43c87b3682bb7c/index.html#L100-L107

`vfile` here isn't failing at runtime anymore.

But it does not work for local pre-bundled dependencies

https://github.com/swandir/vite-browser-external-pnp/blob/1551d40a3cfb8f59915e43f23a43c87b3682bb7c/vite.config.js#L3

This module is loaded in the browser:
https://github.com/swandir/vite-browser-external-pnp/blob/1551d40a3cfb8f59915e43f23a43c87b3682bb7c/dep/node.js#L4

despite being disabled

https://github.com/swandir/vite-browser-external-pnp/blob/1551d40a3cfb8f59915e43f23a43c87b3682bb7c/dep/package.json#L4